libraries/networking/src/ResourceCache.cpp

```diff
@@ -367,63 +364,53 @@ void Resource::handleDownloadProgress(uint64_t bytesReceived, uint64_t bytesTota
 
 void Resource::handleReplyFinished() {
     Q_ASSERT(_request);
-
+    
+    ResourceCache::requestCompleted(this);
+    
     auto result = _request->getResult();
     if (result == ResourceRequest::Success) {
         _data = _request->getData();
         qDebug() << "Request finished for " << _url << ", " << _activeUrl;
-
-        _request->disconnect(this);
-        _request->deleteLater();
-        _request = nullptr;
```

Comment:
My one concern with moving this to the bottom is that a subsequent call relies on them having been cleaned up. There was a specific case with NetworkGeometry when handling a `.fst` it would reuse the Resource for downloading the mesh. I'm not sure if this is still an issue.